sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+        .catch(sqlError => reject(sqlError));
+      } else {
+        resolve('false');
+      }
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function createFileManagerTable(isFileTableFirstInitialize) {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      if (!isFileTableFirstInitialize) {
+        entitiesCreator.createFileManagerTable()
+        .then(success => resolve(true))
+        .catch(sqlError => reject(sqlError));
+      } else {
+        resolve('false');
+      }
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function creationOfTables() {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      initCreateTable()
+      .then(() =>
+        query.isTableExistsInDatabase(constants.ENTITIES_USER)
+        .then(isUserTableFirstInitialize =>
+          createUserTable(isUserTableFirstInitialize)
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+          )
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() =>
+        query.isTableExistsInDatabase(constants.ENTITIES_ORGANIZATION)
+        .then(isOrganizationTableFirstInitialize =>
+          createOrganizationTable(isOrganizationTableFirstInitialize))
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() => query.isTableExistsInDatabase(constants.ENTITIES_FOLDER)
+        .then(isFolderTableFirstInitialize =>
+          createFolderTable(isFolderTableFirstInitialize))
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() => query.isTableExistsInDatabase(constants.ENTITIES_FILE_MANAGER)
+        .then(isFileTableFirstInitialize =>
+          createFileManagerTable(isFileTableFirstInitialize)
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError)))
+        .catch(sqlError => reject(sqlError))
+      .then(isSuccess => resolve(isSuccess))
+      .catch(sqlError => reject(sqlError))
+      );
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+export default class storage {
+  static constructor() {}
+
+  static initializeDatabase() {
+    if (typeof Promise === 'function') {
+      return new Promise((resolve, reject) => {
+        entitiesCreator.initializeDatabase(databaseName)
+        .then(step1 => creationOfTables()
+          .then(isFirstInstance => isFirstInstance)
+          .catch(sqlErr => reject(sqlErr)))
+        .then((isFirstInstance) => {
+          if (typeof isFirstInstance === 'boolean') {
+            return entitiesCreator.fillUpDefaultData()
+            .then(isFilledUp => resolve(isFilledUp))
+            .catch(sqlErr => reject(sqlErr));
+          } else {
```

Comment:
Curious to know, will there be cases where `isFirstInstance` is not a boolean? What is the possible scenario?

If I read correctly, if `isFirstInstance` is not a `boolean` that will imply that the database was previously created, thus there isn't a need to fill up the table?